src/WebJobs.Extensions.DurableTask/Listener/DurableTaskScaleMonitor.cs

```diff
@@ -0,0 +1,149 @@
+using Microsoft.WindowsAzure.Storage;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    internal sealed class DurableTaskScaleMonitor : IScaleMonitor<DurableTaskTriggerMetrics>
+    {
+        private readonly string functionId;
+        private readonly FunctionName functionName;
+        private readonly string hubName;
+        private readonly string storageConnectionString;
+        private readonly ScaleMonitorDescriptor scaleMonitorDescriptor;
+        private readonly Lazy<DisconnectedPerformanceMonitor> performanceMonitor;
+        private readonly EndToEndTraceHelper traceHelper;
+
+        public DurableTaskScaleMonitor(string functionId, FunctionName functionName, string hubName, string storageConnectionString, EndToEndTraceHelper traceHelper)
+        {
+            this.functionId = functionId;
+            this.functionName = functionName;
+            this.hubName = hubName;
+            this.storageConnectionString = storageConnectionString;
+            this.performanceMonitor = new Lazy<DisconnectedPerformanceMonitor>(() => new DisconnectedPerformanceMonitor(this.storageConnectionString, this.hubName));
+            this.scaleMonitorDescriptor = new ScaleMonitorDescriptor($"{this.functionId}-DurableTaskTrigger-{this.hubName}".ToLower());
+            this.traceHelper = traceHelper;
+        }
+
+        public ScaleMonitorDescriptor Descriptor
+        {
+            get
+            {
+                return this.scaleMonitorDescriptor;
+            }
+        }
+
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await this.GetMetricsAsync();
+        }
+
+        public async Task<DurableTaskTriggerMetrics> GetMetricsAsync()
+        {
+            DurableTaskTriggerMetrics metrics = new DurableTaskTriggerMetrics();
+
+            // Durable stores its own metrics, so we just collect them here
+            PerformanceHeartbeat heartbeat = null;
+            try
+            {
+                heartbeat = await this.performanceMonitor.Value.PulseAsync();
+            }
+            catch (StorageException e)
+            {
+                string error = string.Empty;
+                var webException = e.InnerException as WebException;
+                if (webException != null && webException.Status == WebExceptionStatus.ProtocolError)
+                {
+                    string statusCode = ((HttpWebResponse)webException.Response).StatusCode.ToString();
+                    string statusDesc = ((HttpWebResponse)webException.Response).StatusDescription;
+                    error = $"Durable Functions trigger error during PulseAsync: {statusCode}: {statusDesc}";
+                }
+                else if (webException != null && webException.Status == WebExceptionStatus.NameResolutionFailure)
+                {
+                    error = $"Durable Functions trigger storage error during PulseAsync: {webException.Message}";

```

Comment:
How do we know this is a storage error?

Reply:
The storage error message I took from `ScaleController` and added the "during PulseAsync", I'm not sure how that determination is made other than it being of type `StorageException`. My guess is that the `NameResolutionFailure` means the storage account wasn't found? 

The "trigger error" I just realized I added the "error" part so I've fixed it to match `ScaleController`.